sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -26,26 +50,82 @@ class KlaviyoPushService : FirebaseMessagingService() {
         fun getCurrentPushToken(): String {
             return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
         }
+
+        /**
+         * Logs an $opened_push event for a remote notification that originated from Klaviyo
+         *
+         * TODO customer and event properties should both be optional, once SDK is tracking customer state internally
+         *
+         * @param pushPayload The data attributes of the push notification payload
+         * @param customerProperties Profile with which to associate the event
+         * @param eventProperties Optional additional properties for the event
+         */
+        fun handlePush(
+            pushPayload: Map<String, String>,
+            customerProperties: KlaviyoCustomerProperties,
+            eventProperties: KlaviyoEventProperties? = null
+        ) {
+            if (pushPayload["origin"] != "Klaviyo") {
+                return
+            }
+
+            val properties = eventProperties ?: KlaviyoEventProperties()
+            properties.addCustomProperty("push_token", getCurrentPushToken())
+            pushPayload.forEach { (key, value) ->
+                properties.addCustomProperty(key, value)
+            }
```

Comment:
similar to how track internally appends the anonymous ID, should we also internally append the push token?

Reply:
I wouldn't think we want push token appended to _all_ events that are tracked though, so that is why I put it here. `handlePush` is kind of just a special alias to track specific to the $opened_push event.